fossunited/chapters/doctype/foss_chapter_event/foss_chapter_event.py

```diff
@@ -104,8 +104,30 @@ def copy_team_members(self):
             )
 
     def before_save(self):
+        if self.has_value_changed("status"):
+            self.update_published_status()
         self.set_route()
 
+    def update_published_status(self):
+        if self.status == "Draft" or self.status == "Cancelled":
+            self.is_published = 0
+            return
+
+        if self.status == "Live" or self.status == "Approved":
+            self.is_published = 1
+            return
+
+        if self.status == "Concluded":
+            self.is_published = 1
+            self.handle_concluded_event()
+
+    def handle_concluded_event(self):
+        event_permalink = self.event_permalink
+        self.event_permalink = f"{event_permalink}-archive-{frappe.generate_hash(length=8)}"
```

Comment:
What is this for ?

Reply:
as mentioned in the PR description, it is to help archive the events. Permalinks will be changed to end with `-archive-<hash>` which will free up the existing permalink that was being used.